constraint/pkg/client/drivers/k8scel/driver.go

```diff
@@ -33,166 +44,382 @@ package k8scel
-// 		// it doesn't scope visibility.
-
-// 		// template name is the lowercase of its kind
-// 		validator := d.validators[strings.ToLower(constraint.GetKind())]
-// 		if validator == nil {
-// 			return nil, nil, fmt.Errorf("unknown constraint template validator: %s", constraint.GetKind())
-// 		}
-// 		versionedAttr := &generic.VersionedAttributes{
-// 			Attributes:         typedReview,
-// 			VersionedKind:      typedReview.GetKind(),
-// 			VersionedOldObject: typedReview.GetOldObject(),
-// 			VersionedObject:    typedReview.GetObject(),
-// 		}
-// 		// FRICTION: member variables of `decision` are private, which makes it impossible to consume the results. I got around this by forking the apiserver code
-// 		decisions, err := validator.Validate(versionedAttr, constraint)
-// 		if err != nil {
-// 			return nil, nil, err
-// 		}
-
-// 		enforcementAction, found, err := unstructured.NestedString(constraint.Object, "spec", "enforcementAction")
-// 		if err != nil {
-// 			return nil, nil, err
-// 		}
-// 		if !found {
-// 			enforcementAction = apiconstraints.EnforcementActionDeny
-// 		}
-// 		for _, decision := range decisions {
-// 			if decision.Action == validatingadmissionpolicy.ActionDeny {
-// 				results = append(results, &types.Result{
-// 					Target:            target,
-// 					Msg:               decision.Message,
-// 					Constraint:        constraint,
-// 					EnforcementAction: enforcementAction,
-// 				})
-// 			}
-// 		}
-// 	}
-// 	return results, nil, nil
-// }
-
-// func (d *Driver) Dump(ctx context.Context) (string, error) {
-// 	return "", nil
-// }
-
-// // FRICTION we should not need to create mocks to use this library offline... this is used for version conversion, which
-// // cannot be done reliably offline and is superfluous for audit, as audit scrapes all versions anyway
-// // Currently, creating a mock that returns nil and/or errors risks the code breaking on library upgrade. It would be good
-// // to have a contract here.
-// var _ admission.ObjectInterfaces = &mockObjectInterfaces{}
-
-// type mockObjectInterfaces struct{}
-
-// type mockObjectCreater struct{}
-
-// func (m *mockObjectCreater) New(gvk schema.GroupVersionKind) (runtime.Object, error) {
-// 	return nil, errors.New("OBJECT CREATOR NOT IMPLEMENTED")
-// }
-
-// func (m *mockObjectInterfaces) GetObjectCreater() runtime.ObjectCreater { return &mockObjectCreater{} }
-
-// func (m *mockObjectInterfaces) GetObjectTyper() runtime.ObjectTyper { return nil }
-
-// func (m *mockObjectInterfaces) GetObjectDefaulter() runtime.ObjectDefaulter { return nil }
-
-// func (m *mockObjectInterfaces) GetObjectConvertor() runtime.ObjectConvertor { return nil }
-
-// func (m *mockObjectInterfaces) GetEquivalentResourceMapper() runtime.EquivalentResourceMapper {
-// 	return nil
-// }
+//   Other friction points are commented with the keyword FRICTION.
+
+const (
+	Name = "K8sValidation"
```

Comment:
I think we called this `K8sNativeValidation` in the Kubecon talk to avoid confusion, do we want to keep that?

Reply:
SGTM, IMO we are free to change any of this after-the-fact (prototype), but making the change now.